plugins/guests/tinycore/cap/change_host_name.rb

```diff
@@ -4,7 +4,6 @@ module Cap
       class ChangeHostName
         def self.change_host_name(machine, name)
           if !machine.communicate.test("hostname | grep '^#{name}$'")
-            machine.communicate.sudo("sh -c 'echo \"#{name}\" > /etc/hostname'")
```

Comment:
@sethvargo nope, this is because two things :
* boot2docker is RAMDISK based (exceptions exists but for /var/lib/docker and /var/lib/boot2docker only)
* The bootscript of boot2docker will rewrite this everytime, before ssh daemon has been started :  https://github.com/boot2docker/boot2docker/blob/master/rootfs/rootfs/bootscript.sh#L22 (Warning : point to the b2d master, curent tag is 1.6.0 at the date of this comment) 
which call https://github.com/boot2docker/boot2docker/blob/master/rootfs/rootfs/etc/rc.d/hostname

=> So the scenariis here are simpler than with "persisted" guest like debian or redhats